hypothesis-python/tests/array_api/test_strategies_namespace.py

```diff
@@ -43,7 +55,28 @@ def test_caching(xp, monkeypatch):
     assert len(array_api._args_to_xps) == 0
 
 
-@pytest.mark.filterwarnings("ignore::hypothesis.errors.HypothesisWarning")
+@pytest.mark.parametrize(
+    "api_version1, api_version2", [(None, "2021.12"), ("2021.12", None)]
+)
+def test_inferred_namespace_is_cached_seperately(
+    api_version1, api_version2, monkeypatch
+):
+    """Results from inferred versions do not share the same cache key as results
+    from specified versions."""
```

Comment:
I *do* want them to share this, such that if you pass an explicit version and then infer you get the same namespace object.

Reply:
Oops I can't read :facepalm: pushed a fix